managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -3738,6 +3751,35 @@ public void setEntriesAddedCounter(long count) {
 
     private static final Logger log = LoggerFactory.getLogger(ManagedLedgerImpl.class);
 
+    @Override
+    public CompletableFuture<Void> asyncTruncate() {
+
+        final List<CompletableFuture<Void>> futures = Lists.newArrayList();
+        for (ManagedCursor cursor : cursors) {
+            final CompletableFuture<Void> future = new CompletableFuture<>();
+            cursor.asyncClearBacklog(new AsyncCallbacks.ClearBacklogCallback() {
```

Comment:
Why need to clear backlog?

Reply:
If it doesn't clearly backLog, it will be trim instead of truncate. So I do it.

Reply:
Clear backlog will lose unacknowledged messages?

Reply:
The original request is delete retained message, retained message is acknowledged message, message in backlog is unacknowledged message.
 There are dedicated command to clear whole backlog or expire certain messages in backlog, for this command we should just delete retained message so we shouldn't clear backlog here.

Reply:
Thank you very much for your comments. I think ‘truncate' is more like deleting all the messages than deleting the consumed messages. And I want to add a 'trim' function to the next PR.